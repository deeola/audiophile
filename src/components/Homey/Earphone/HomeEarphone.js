import React from "react";
import DescriptionText from "../Z7speakers/DescriptionText";
import {
  ContDiv,
  HomeEarphoneContainer,
} from "./HomeEarphoneStyling";

const HomeEarphone = () => {

  return (
    <HomeEarphoneContainer data-testid='earphone'>
      <ContDiv img='left'>
      </ContDiv>
      <ContDiv>
        <DescriptionText title={"YX1 EARPHONES"} link="/yx1" />
      </ContDiv>
    </HomeEarphoneContainer>
  );
};

export default HomeEarphone;
