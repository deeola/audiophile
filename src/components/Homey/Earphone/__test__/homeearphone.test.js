import reactDom from "react-dom";
import HomeEarphone from "../HomeEarphone";
import { render, cleanup } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

it("renders div correctly", () => {
  const div = document.createElement("div");

  reactDom.render(
    <BrowserRouter>
      <HomeEarphone />
    </BrowserRouter>,
    div
  );
});

it("has two div elements", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <HomeEarphone />
    </BrowserRouter>
  );
  expect(getByTestId("earphone").childElementCount).toBe(2);
});

//snapshot

it("matches  snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <HomeEarphone />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
