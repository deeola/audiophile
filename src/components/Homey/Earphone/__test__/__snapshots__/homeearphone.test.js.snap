// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches  snapshot 1`] = `
<section
  className="sc-eCYdqJ kjZOWd"
  data-testid="earphone"
>
  <div
    className="sc-jSMfEi fwShuz"
  />
  <div
    className="sc-jSMfEi iUnqkz"
  >
    <div
      className="sc-dkzDqf"
    >
      <h4
        className="sc-hKMtZM cXPvZu"
        data-testid="title"
      >
        YX1 EARPHONES
      </h4>
      <a
        className="sc-bczRLJ cwyJdT"
        color="secondary"
        hover="secondary"
        href="/yx1"
        onClick={[Function]}
      >
        SEE PRODUCT
      </a>
    </div>
  </div>
</section>
`;
